arm/proofs/instruction.ml

```diff
@@ -1931,6 +1948,193 @@ let arm_STP = define
             else (=))
          else ASSIGNS entirety) s`;;
 
+(** LD1 with 1 register is equivalent to simply loading the whole word *)
+let arm_LD1_1 = define
+  `arm_LD1_1 (Rt:(armstate, N word)component) Rn off = 
+    \s. let address = read Rn s in
+        let eaddr = word_add address (offset_address off s) in
+        (if (Rn = SP ==> aligned 16 address) /\
+            (offset_writesback off ==> orthogonal_components Rt Rn)
+         then
+           Rt := read (memory :> wbytes eaddr) s ,,
+           (if offset_writesback off
+            then Rn := word_add address (offset_writeback off)
+            else (=))
+         else ASSIGNS entirety) s`;;
```

Comment:
This can be a naive question, but if arm_LD1_1 is simply loading the whole word can using existing `arm_ldst*` be used? This will make the diff of this patch smaller & have less maintenance cost.

Reply:
In general, LD1/ST1 is not the same as arm_LDR/STR because they are also defined for 2,3,4 registers. However, I don't see any use of those cases in my program, so I only modeled the case for 1 register. For 1 register, it is exactly the same as LDR and STR. I merged the definitions in the following commit. Let me know if you like it :)

Reply:
The two updates looked really nice, thanks..! :)